File: src/main/java/org/academy/internal/client/app/music/backend/decoder/AudioStream.java
package org.academy.internal.client.app.music.backend.decoder;

import java.io.IOException;
import java.nio.ShortBuffer;

public interface AudioStream extends AutoCloseable {
    int getChannels();

    int getSampleRate();

    long getTotalSamples();

    int read(ShortBuffer pcmBuffer) throws IOException;

    void seek(long sampleFrame) throws IOException;

    @Override
    void close();
}

File: src/main/kotlin/org/academy/internal/client/app/music/decoder/AudioStream.kt
package org.academy.internal.client.app.music.decoder

import java.nio.ShortBuffer

interface AudioStream : AutoCloseable {
    val channels: Int
    val sampleRate: Int
    val totalSamples: Long

    fun read(pcmBuffer: ShortBuffer): Int

    fun seek(sampleFrame: Long)

    override fun close()
}